store/store.go

```diff
@@ -494,33 +494,56 @@ func (bs *BlockStore) Close() error {
 
 //-----------------------------------------------------------------------------
 
+// Key prefixes for different types of data
+const (
+	// Key prefixes
+	blockMetaPrefix  = "H"   // Height -> BlockMeta
+	blockPartPrefix  = "P"   // Height + index -> Block Part
+	blockCommitPrefix = "C"  // Height -> Commit
+	seenCommitPrefix = "SC"  // Height -> Seen Commit
+	blockHashPrefix  = "BH"  // Hash -> Height
+	txHashPrefix     = "TH"  // Hash -> TxInfo
+	stateKey        = "blockStore" // Fixed key for block store state
+)
```

Comment:
[question] what motivated this change? Can it be upstreamed to CometBFT so that we can pull it here? None of this seems specific to Celestia.

Ref: https://github.com/celestiaorg/celestia-core?tab=readme-ov-file#contributing